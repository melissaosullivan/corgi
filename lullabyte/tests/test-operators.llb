void main()
{
	print(3+5);
	print(9-2);
	print(4*6);
	print(10/2);
	print(4%2);
	print(true || true);
	print(true || false);
	print(false && true);
	print(true && true);
	print(3<5);
	print(8<=2);
	print(19>3);
	print(3>=5);
	print(4==4);
	print(1!=2);
	print(true==true);
	print(false==true);
	print(1.2+2.5);
	print(9.0-2.4);
	print(0.5*0.5);
	print(2.5/0.25);
	print(4.4==2.4);
	print(4.000!=2.04);
	print(2.20<4.9);
	print(0.0>1.3);
	print(190.3>=190.3);
	print(100.2<=3.3);
	print(!true);
	print(-4);
	print(-4.4);
	print(2+2.2);
	print(3.5+2);
	print(3.5-2);
	print(1-.5);
	print(6.1*3);
	print(3*4.2);
	print(1/4.0);
	print(8.2/4);
	print(9.0==9);
	print(9==9.1);
	print(1.0!=1);
	print(2!=1.1);
	print(1.1<3);
	print(4<3.2);
	print(4>5.1);
	print(5.1>4);
	print(10.<=10);
	print(11<=4.1);
	print(2.3>=6);
	print(9>=8.2);
	print(1+C0);
	print(Eb5+3);
	print(60-C0);
	print(F#3-2);
	print(5*F#0);
	print(A0*10);
	print(24/C1);
	print(D6/2);
	print(100%A0);
	print(E8%10);
	print(28==E2);
	print(A#4==33);
	print(21!=A1);
	print(B5!=40);
	print(14<A#5);
	print(B4<12);
	print(2>D3);
	print(E9>40);
	print(7<=B3);
	print(A3<=2);
	print(5>=G9);
	print(G8>=6);
	print(C3==C3);
	print(C5!=C5);
	print(B1<B2);
	print(B1>B2);
	print(E5<=E6);
	print(E5>=E6);
	print(4*|C5|:.25:100);
	print(.5*|C2|:.5:100);
	print(|D4, C9|:.5:100*2);
	print(|A5, D3|:1.:100*.25);
	print([|G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100]*1);
	print([|G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100]*3);
	print([4, 6, 10]*4);
	print([3., .5]*2);
	print(1*[|G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100]);
	print(3*[|G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100]);
	print(4*[4, 6, 10]);
	print(2*[3., .5]);
}
