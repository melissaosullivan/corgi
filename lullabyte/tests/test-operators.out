8
7
24
5
0
true
true
false
true
true
false
true
false
true
true
true
false
3.7
6.6
0.25
10.
false
true
true
false
true
false
false
-4
-4.4
4.2
5.5
1.5
0.5
18.3
12.6
0.25
2.05
true
false
false
true
true
false
false
true
true
false
false
true
C#0
F#5
C5
E3
F#2
F#7
D0
C#3
C#0
C0
true
false
false
true
true
false
false
true
true
false
false
true
true
false
true
false
true
false
|C5|:1.:100
|C2|:0.25:100
|D4, C9|:1.:100
|A5, D3|:0.25:100
[|G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100]
[|G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100, |G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100, |G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100]
[4, 6, 10, 4, 6, 10, 4, 6, 10, 4, 6, 10]
[3., 0.5, 3., 0.5]
[|G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100]
[|G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100, |G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100, |G2, C1|:0.25:100, |D1|:0.25:100, |E1|:0.25:100, |F1|:0.25:100]
[4, 6, 10, 4, 6, 10, 4, 6, 10, 4, 6, 10]
[3., 0.5, 3., 0.5]
